src/cipher.rs

```diff
@@ -1161,9 +1161,29 @@ impl Castable for rustls_server_cert_verifier {
 }
 
 impl rustls_server_cert_verifier {
+    /// Create a new server certificate verifier that uses the system's root store and WebPKI via
+    /// [`rustls-platform-verifier`][].
+    ///
+    /// The verifier can be used in several `rustls_client_config` instances and must be freed by
+    /// the application using `rustls_server_cert_verifier_free` when no longer needed.
+    ///
+    /// [`rustls-platform-verifier`]: https://github.com/rustls/rustls-platform-verifier
+    #[no_mangle]
+    pub extern "C" fn rustls_platform_server_cert_verifier(
+        verifier_out: *mut *mut rustls_server_cert_verifier,
+    ) -> rustls_result {
```

Comment:
Since the only way this function can return an error is on invalid inputs (i.e. `try_mut_from_ptr_ptr!` returns an error), it's better to just return the new pointer: `-> *mut rustls_server_cert_verifier`. From the README:

> ## NULL
>
> The library checks all pointers in arguments for NULL and will return an error
> rather than dereferencing a NULL pointer. For some methods that are infallible
> except for the possibility of NULL (for instance
> `rustls_connection_is_handshaking`), the library returns a convenient
> type (e.g. `bool`) and uses a suitable fallback value if an input is NULL.

Reply:
Good point, done :+1: